Added Notifications settings for viewers

JanSuklje opened the pull request:
## Proposed change
<!--
  Thank you!

  If you're introducing a new feature or significantly refactoring existing functionality,
  we encourage you to start a discussion first. This helps ensure your idea aligns with
  Frigate's development goals.

  Describe what this pull request does and how it will benefit users of Frigate.
  Please describe in detail any considerations, breaking changes, etc. that are
  made in this pull request.
-->

Users with Viewer role can now see Notifications settings view and register their devices.
Changed the view for this role so it contains only the necessary information alongside the button for registering to notifications.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code
- [ ] Documentation Update

## Additional information


## Checklist

<!--
  Put an `x` in the boxes that apply.
-->

- [x] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [ ] UI changes including text have used i18n keys and have been added to the `en` locale.
- [ ] The code has been formatted using Ruff (`ruff format frigate`)

hawkeye217 commented:
Thanks for this PR. I think this should be targeted at `master` instead of `dev` as we'd like to bring it into 0.16.1 instead of 0.17.

Also, I think it would be preferable to selectively add `isAdmin` checks around the other elements rather than reproducing a separate return block for viewers that duplicates the same code for the title, description, and register/unregister button.

Curious what the other maintainers have thoughts.

NickM-27 commented:
> Also, I think it would be preferable to selectively add isAdmin checks around the other elements rather than reproducing a separate return block for viewers that duplicates the same code for the title, description, and register/unregister button.

Yes, definitely agree

NickM-27 commented:
You can use the `cn()` function to pass a classname with overrides when the user is view only, that way the columned layout can be overridden

JanSuklje committed:
Refactored Viewer role Notifications settings

 - now each individual element is shown if allowed by role, instead of having multiple return statement for each role

hawkeye217 reviewed (approved):
Looks great, thanks for your contribution.

JanSuklje commented:
I don't want to open a new PR unnecessarily so I will just ask here. I don't like how there is so much padding at the bottom so I tweaked it a bit. Would this change be ok?

This is my first contribution to Frigate so im not sure how freely we can change the UI and if we should follow some general patterns (noticed that some settings have the same padding at the bottom but this one has the least elements so it stands out the most).

Current:
<img width="1920" height="882" alt="image" src="https://github.com/user-attachments/assets/a9bd4b8d-e16d-4ced-b384-9636345b3bd4" />

Proposed:
<img width="1920" height="883" alt="image" src="https://github.com/user-attachments/assets/cc065e20-77c8-40d5-b1d3-3d8cb090059a" />
